### DISCUSSION TITLE
[Done] Async Events

### DISCUSSION THREAD

**Discussion by MrNemo64** [ArcanePlugins/Treasury#115] [Ideas]
Right now the events don't specify if they are sync or async but the economy response is defined to be async.
When a consumer triggers an operation that may trigger an event, does the event run sync or async? Should we consider this?

**Comment by lokka30** [comment]
My current opinion: I think the economy provider should be able to choose which thread they would like to use. Ideally not the main thread, but I don't mind if they choose it.

**Reply by MrNemo64** [reply]
Maybe we should add an `isAsynchronous()` method?

**Reply by lokka30** [reply]
Is there an easy way for us to tell if the event is on the main thread?

**Reply by MrNemo64** [reply]
There is [`Bukkit#isPrimaryThread()`](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Bukkit.html#isPrimaryThread()) but I don't know if it checks that you are on the main thread or on a thread that is synchronized with the main thread

**Comment by MrIvanPlays** [comment]
Thats an easy one, add the async parameter into the constructors. An economy provider may choose to call them sync.

**Reply by MrNemo64** [reply]
That was my first idea and probably the best one

**Reply by lokka30** [reply]
Sounds great. Assuming it still has the `isAsynchronous` method.

**Reply by MrNemo64** [reply]
> Sounds great. Assuming it still has the `isAsynchronous` method.

Yes, and would return the boolean

**Reply by MrIvanPlays** [reply]
Bukkit event already has an async parameter. Just a constructor parameter and calling the super constructor is enough.

**Comment by MrIvanPlays** [comment]
https://github.com/lokka30/Treasury/commit/5a9b94a87ec115c5ebdb17f9dd013223708279e7
